src/servlet/DeleteServlet.java

```diff
@@ -0,0 +1,97 @@
+package servlet;
+
+import java.io.IOException;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import bean.CorrectAnswersBean;
+import bean.QuestionsBean;
+import dao.CorrectAnswersDao;
+import dao.QuestionsDao;
+
+/**
+ * Servlet implementation class DeleteServlet
+ */
+@WebServlet("/DeleteServlet")
+public class DeleteServlet extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+
+    /**
+     * @see HttpServlet#HttpServlet()
+     */
+    public DeleteServlet() {
+        super();
+        // TODO Auto-generated constructor stub
+    }
+
+	/**
+	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		// TODO Auto-generated method stub
+		//formの値を日本語にする
+		request.setCharacterEncoding("UTF-8");
+
+		try {
+			//リクエストパラメーターの取得
+			//変数に保存
+			String str_question_id = request.getParameter("questionId");
+			String question = request.getParameter("question");
+			String[] str_answer_id = request.getParameterValues("answerId");
+			String[] str_questions_id = request.getParameterValues("questionsId");
+			String[] answer = request.getParameterValues("answer");
+
+			// intに変換して取得
+			int question_id = Integer.parseInt(str_question_id);
+
+			//リクエストパラメーターから受け取った値をセッタを使って書き込む
+			QuestionsBean qb = new QuestionsBean();
+			qb.setId(question_id);
+			qb.setQuestion(question);
+
+			//配列で受け取る
+			CorrectAnswersBean cab = new CorrectAnswersBean();
+
+			//データベースに追加するデータを保持するQuestionsとAnswersオブジェクトを作成
+			//QuestionsDaoのインスタンスオブジェクトを生成（インスタンス化）
+			QuestionsDao qdao = new QuestionsDao();
+			CorrectAnswersDao cadao = new CorrectAnswersDao();
+		
+			//QuestionsDAOをdelete
+			qdao.delete(qb); //delete
+
+			int[] answer_ids = new int[str_answer_id.length];
+			int[] questions_id = new int[str_questions_id.length];
+
+			//配列の長さ分繰り返してdelete
+			for (int i = 0; i < answer.length; i++) {
+				//数値型に変換
+				answer_ids[i] =  Integer.parseInt(str_answer_id[i]);
+				questions_id[i] =  Integer.parseInt(str_questions_id[i]);
+				
+				//question.idとcorrect_answers.question_idが同じ場合にdeleteする
+				if (answer_ids[i] == questions_id[i]) {
```

Comment:
これcorrect_answersのidとcorrect_answersのquestion_idが一致してるかをみてませんか？
データによっては一致するものもありそうですが
本来やるべきはコメントに書いてることですね。なのでこうなりそう。
```suggestion
				if (question_id == questions_id[i]) {
```

Reply:
確認不足でした、、
ありがとうございます！